lib/enhance-assert.js

```diff
@@ -8,8 +8,8 @@ module.exports.PATTERNS = [
 	't.false(value, [message])',
 	't.is(value, expected, [message])',
 	't.not(value, expected, [message])',
-	't.same(value, expected, [message])',
-	't.notSame(value, expected, [message])',
+	't.deepEqual(value, expected, [message])',
+	't.notDeepEqual(value, expected, [message])',
```

Comment:
The `power-assert` enhancement actually has callbacks (`onSuccess`, `onFailure`) that are called with the result of each assertion. That is where we implement assertion tracking.

I think you should probably keep `t.same` and `t.notSame` in the patterns until they are officially removed. Otherwise this will break assertionCount for them. Maybe add a test that ensures the assertion count is tracked for those (which we will remove when we drop support).